src/services/__tests__/scheduler-service.test.ts

```diff
@@ -0,0 +1,64 @@
+import 'reflect-metadata'
+import { SchedulerService } from '../scheduler-service'
+import { config } from 'node-config-ts'
+import { container } from 'tsyringe'
+import { jest } from '@jest/globals'
+
+describe('scheduler service', () => {
+  jest.setTimeout(20000)
+  beforeAll(async () => {
+    container.registerInstance('config', Object.assign({}, config, { schedulerIntervalMS: 3000 }))
+    container.registerSingleton('schedulerService', SchedulerService)
+  })
+
+  test('will run the task repeatedly', (done) => {
+    const numberOfRunsBeforeDone = 3
+    const schedulerService = container.resolve<SchedulerService>('schedulerService')
+
+    let count = 0
+    const task = jest.fn(() => {
+      if (count === numberOfRunsBeforeDone) {
+        schedulerService.stop()
+        runChecks()
+      }
+
+      count = count + 1
+      return Promise.resolve()
+    })
+
+    schedulerService.start(task)
+    const runChecks = () => {
+      // the task runs once right at the start before running every X seconds
+      expect(task.mock.calls.length).toEqual(numberOfRunsBeforeDone + 1)
+      done()
+    }
```

Comment:
doesn't this test need to wait for the schedulerService to shut down?  otherwise couldn't the test finish while the scheduler is still running, before `runChecks` even gets called?

What happens if you put a purposefully failing assertion in runChecks?  Would be interesting to see how jest handles that assertion failing if the test has already completed running.

Reply:
You could change `schedulerService.start()` to return a promise that resolves when the service shuts down, and await that promise at the end of the test.

Reply:
OH!  I see now the `done` argument.  I didn't realize that you could tell jest not to consider a test finished until you call a given function.  Cool.  Maybe then just add a comment as the very last line of the test `// test doesn't complete until 'done()' is called` so it's clearer to future readers.

Reply:
I added the comment =)